src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -182,6 +181,41 @@ private void OnRequestShowInCanvasSearch(object param)
         public bool IsSnapping { get; set; }
 
         /// <summary>
+        /// Gets the collection of Dynamo-specific preferences.
+        /// This is used when serializing Dynamo preferences in the View block of Graph.Json.
+        /// </summary>
+        [JsonProperty("Dynamo")]
+        public DynamoPreferencesData DynamoPreferences
+        {
+            get
+            {
+              bool hasRunWithoutCrash = false;
+              string runType = RunType.Manual.ToString();
+              string runPeriod = RunSettings.DefaultRunPeriod.ToString();
+              HomeWorkspaceModel homeWorkspace = Model as HomeWorkspaceModel;
+              if (homeWorkspace != null)
+              {
+                hasRunWithoutCrash = homeWorkspace.HasRunWithoutCrash;
+                runType = homeWorkspace.RunSettings.RunType.ToString();
+                runPeriod = homeWorkspace.RunSettings.RunPeriod.ToString();
+              }
+
+              bool isVisibleInDynamoLibrary = true;
+              CustomNodeWorkspaceModel customNodeWorkspace = Model as CustomNodeWorkspaceModel;
+              if (customNodeWorkspace != null)
+                isVisibleInDynamoLibrary = customNodeWorkspace.IsVisibleInDynamoLibrary;
+
+              return new DynamoPreferencesData(
+                Model.ScaleFactor,
+                hasRunWithoutCrash,
+                isVisibleInDynamoLibrary,
+                Model.WorkspaceVersion.ToString(),
```

Comment:
hmmm, I think when we save we actually want to save the current DynamoModel version not the old workspace version right?